src/XrdSciTokens/XrdSciTokensAccess.cc

```diff
@@ -353,6 +353,10 @@ class XrdAccRules
             if ((oper == rule.first) && !path.compare(0, rule.second.size(), rule.second, 0, rule.second.size())) {
                 return true;
             }
+            // according to WLCG token specs, allow creation of required superfolders for a new file if requested
+            if ((oper == AOP_Stat || oper == AOP_Mkdir) && !rule.second.compare(0, path.size(), path, 0, path.size()) &&  rule.second.size() >= path.length()  ) {
```

Comment:
This would allow a `mkdir` operation using a read-only token, correct?  How about something like this:
```suggestion
            if ((oper == rule.first) && (oper == AOP_Stat || oper == AOP_Mkdir) && rule.second.size() >= path.length() && !rule.second.compare(0, path.size(), path, 0, path.size()) && (rule.second.size() == path.length() || rule.second[path.length()] == '/') ) {
```

The last part of the conditional:

```
(rule.second.size() == path.length() || rule.second[path.length()] == '/')
```

mandates either the two strings are the same _xor_, as the ACL path is longer, the next character in the ACL path is '/'.  That is:

- requested: mkdir /foo, acl: mkdir /foobar/baz => FAIL
- requested: mkdir /foobar, acl: mkdir /foobar => OK
- requested: mkdir /foobar/baz, acl: mkdir /foobar/baz => OK
- requested: mkdir /, acl: mkdir /foobar/baz => FAIL

Reply:
thanks Brian, I've updated the PR with your suggestion